invenio_users_resources/services/users/service.py

```diff
@@ -74,7 +77,10 @@ def read(self, identity, id_):
         # resolve and require permission
         user = UserAggregate.get_record(id_)
         if user is None:
-            raise LookupError(f"No user with id '{id_}'.")
+
+            raise UserNotFoundError(
+                _("User with id '{id_}' not found.").format(id_=id_)
```

Comment:
yes but we would still expose hidden profiles

Reply:
Valid point, I didn't think about this senario! thank you for pointing it out, the end-user should not know much details when it comes to users and groups, I will fix it now!

Reply:
ping @slint @alejandromumo

Reply:
Since funders/names/affiliations (and others) are vocabularies used for just referencing in bibliographic metadata publicly, they do not expose any personal information which might lead to a user enumeration attack. In this case, a 404 response should be fine.

IIRC the HTTP response code for these errors is configurable per REST API resource, so I think we should aim at providing some sane defaults, i.e. 404 for public resources, and make exceptions for private/personal information with a 403.